src/main/groovy/edu/oregonstate/mist/osuevents/resources/EventsResource.groovy

```diff
@@ -0,0 +1,239 @@
+import javax.ws.rs.Consumes
+import javax.ws.rs.DELETE
+import javax.ws.rs.GET
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.PathParam
+import javax.ws.rs.Produces
+import javax.ws.rs.QueryParam
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+import java.time.format.DateTimeParseException
+
+import static java.util.UUID.randomUUID
+
+@Path('/events/')
+@Produces(MediaType.APPLICATION_JSON)
+class EventsResource extends Resource {
+
+    Logger logger = LoggerFactory.getLogger(EventsResource.class)
+
+    private final EventsDAO eventsDAO
+
+    EventsResource(EventsDAO eventsDAO) {
+        this.eventsDAO = eventsDAO
+    }
+
+    //used to check if client-generated id is a valid UUID
+    private final String uuidRegEx =
+            "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[34][0-9a-fA-F]{3}-[89ab][0-9a-fA-F]{3}-[0-9a-fA-F]{12}"
+
+/**
+ * GET by ID
+ */
+    @GET
+    @Path('{id: [0-9a-zA-Z-]+}')
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getByID(@Auth AuthenticatedUser _,
+                            @PathParam('id') String id) {
+        ResourceObject event = eventsDAO.getById(id)
+        ok(getResultObject(event)).build()
+    }
+
+/**
+ * GET all events
+ */
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response getEvents(@Auth AuthenticatedUser _,
+                              @QueryParam('format') String format) {
+        def events = eventsDAO.getEvents()
+//@TODO add methods for returning csv and ics formats
+//        else if (format == "csv") {
+//
+//        } else if (format == "ics") {
+//
+//        } else {
+//            return badRequest("Invalid format value. Valid formats are csv or ics.").build()
+//        }
+        ok(getResultObject(events)).build()
+    }
+/**
+ * POST an event
+ */
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response createEvent(@Auth AuthenticatedUser _,
+                                @Valid ResultObject newResultObject) {
+        ResourceObject newResourceObject
+        Event newEvent
+
+        try {
+            newResourceObject = newResultObject.data
+            newEvent = newResultObject.data.attributes
+        } catch (GroovyCastException e) {
+            return badRequest(ErrorMessages.unknownFields).build()
+        } catch (Exception e) {
+            logger.error("Exception while calling createEvent", e)
+            return internalServerError(ErrorMessages.unexpectedException).build()
+        }
+
+        //generate id if one wasn't included in the request
+        newResourceObject.id = newResourceObject.id ?: randomUUID() as String
+
+        if (!newResourceObject.id.matches(uuidRegEx) || eventsDAO.getById(newResourceObject.id)) {
+            return conflict().build()
+        }
+
+        //get custom fields and filters ready for DAO
+        String customFieldData = JsonOutput.toJson(newEvent.customFields)
```

Comment:
Malformed json is caught before the method is called, handled by this annotation for POST and PATCH: 
`@Valid ResultObject newResultObject`